output/shared/messagepacker.go

```diff
@@ -0,0 +1,142 @@
+package shared
+
+import (
+	"bytes"
+	"compress/gzip"
+
+	"github.com/relex/gotils/logger"
+	"github.com/relex/slog-agent/base"
+	"github.com/relex/slog-agent/defs"
+	"github.com/relex/slog-agent/util"
+)
+
+// gzipCompressionLevel for ForwardMessage.
+// BestSpeed uses 30% more space and roughly same percentage in time saving
+const gzipCompressionLevel = gzip.BestSpeed
+
+type PackerConfig struct {
+	MsgBufCapacity    int
+	ChunkMaxSizeBytes int
+	ChunkMaxRecords   int
+	ChunkIDSuffix     string
+	UseCompression    bool
+}
+
+// messagePacker writes incoming log messages into a batch (chunk) and encodes them to gzip
+type messagePacker struct {
+	logger            logger.Logger
+	useCompression    bool
+	reusedGzipBuffer  *bytes.Buffer // buffer for gzipWriter for log records
```

Comment:
The original name `reusedGzipBuffer` is misleading as it's used for both gzip and non-compressed form. Might be better to be named like "reusedOutputBuffer" or "chunkOutputBuffer"